omod/src/main/webapp/resources/scripts/components/LabTrackingOrderFactory.js

```diff
@@ -333,10 +347,11 @@ angular.module("labTrackingOrderFactory", [])
                 concept: LabTrackingOrder.concepts.order.conceptUuid,
                 careSetting: labTrackingOrder.careSetting.value,
                 encounter: labTrackingOrder.encounter.value,
-                orderReason: labTrackingOrder.preLabDiagnosis.value,
+                orderReason: typeof labTrackingOrder.preLabDiagnosis === 'object' ? labTrackingOrder.preLabDiagnosis.value : null,
+                orderReasonNonCoded: typeof labTrackingOrder.preLabDiagnosis === 'string' ? labTrackingOrder.preLabDiagnosis : null,
                 instructions: labTrackingOrder.instructions.value,
                 clinicalHistory: labTrackingOrder.clinicalHistory.value,
-                dateActivated: labTrackingOrder.requestDate.value ? labTrackingOrder.requestDate.value : new Date() 
+                dateActivated: labTrackingOrder.requestDate.value ? labTrackingOrder.requestDate.value : new Date()
```

Comment:
```suggestion
                dateActivated: labTrackingOrder.requestDate.value || new Date()
```

Reply:
What you have seems clearer but I'm going to avoid refactoring this part since this code isn't really part of the issue/changes (looks like it just got picked up because a newline or space was added/removed?)  and so won't be covered in testing.